src/HostnameValidator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validates Host name and it's IP address
+ */
+class HostnameValidator extends BaseValidator
+{
+    use Translation;
+
+    /**
+     * Validates domain names against RFC 1034, RFC 1035, RFC 952, RFC 1123, RFC 2732, RFC 2181, and RFC 1123
+     *
+     * @param string $value
+     * @return boolean
+     */
+    public function isValid($value)
+    {
+        $this->clearMessages();
+
+        $asciiHostname = idn_to_ascii($value, 0, INTL_IDNA_VARIANT_UTS46);
```

Comment:
What's the purpose of `INTL_IDNA_VARIANT_UTS46`? The email address validator doesn't use it, should it?

Reply:
This is used to support idn_to_ascii for php version 7.2 as the default `variant` for versions below 7.2 is `INTL_IDNA_VARIANT_2003`